include/libtorrent/socket.hpp

```diff
@@ -81,6 +81,26 @@ namespace libtorrent {
 	using boost::asio::null_buffers;
 #endif
 
+	// this is a silly wrapper for address and endpoint to make their move
+	// constructors be noexcept (because boost.asio is incorrectly making them
+	// potentially throwing). This can be removed once libtorrent uses the
+	// networking TS
+	template <typename T>
+	struct noexcept_move_constructible : T
```

Comment:
this is kind of an unfortunate hack, but the alternative would be to manually spell out a lot of alert type constructors, to work around that the defaulted move constructor can't be noexcept